src/backend/joanie/payment/backends/base.py

```diff
@@ -110,7 +110,7 @@ def _do_on_payment_failure(order):
         Mark the invoice as canceled and cancel the order.
         """
         # - Unvalidate order
-        order.cancel()
+        order.payment_failed()
```

Comment:
the previous name seemed better for an action? the name you propose sounds like a state?

Reply:
Yes but cancel changes the state to `cancelled` not `pending`. I'll look for another name